omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,145 @@
+var app = angular.module("conditionApp",
+    ['uicommons.widget.coded-or-free-text-answer', 'app.restfulServices', 'app.models', 'app.commonFunctionsFactory']);
+
+app.controller("ConditionController", ConditionController);
+ConditionController.$inject = ['$scope', 'RestfulService', 'ConditionModel', 'ConceptModel', 'CommonFunctions'];
+
+function ConditionController($scope, RestfulService, ConditionModel, ConceptModel, CommonFunctions) {
+    var self = this;
+
+    $scope.patientUuid = null;
+    $scope.condition = null;
+
+    // this is required inorder to initialize the Restangular service
+    RestfulService.setBaseUrl('/' + OPENMRS_CONTEXT_PATH + '/ws/rest/emrapi');
+
+    /**
+     * Perform POST save
+     * PS: Make sure the 'Non Coded Uuid' setting (under Settings->Condition List) is set inorder to work with noncoded concepts,
+     * the LOCALE needs to be set (most concepts have 'en')
+     * @type {Function}
+     */
+    self.saveCondition = self.saveCondition || function () {
+        if($scope.condition.onSetDate>$scope.condition.endDate){
+                emr.errorAlert("End Date can't be before Onset Date");
+        }
+        else{
+            var conditions = [];
+            conditions.push($scope.condition);
+            RestfulService.post('condition', conditions, function (data) {
+                //emr.successAlert("conditionlist.updateCondition.success"); Messages not being resolved
+                emr.successAlert("Condition edited successfully")
+                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
+            }, function (error) {
+                //emr.errorAlert("conditionlist.updateCondition.error");
+                emr.errorAlert("Error Saving condition");
+            });
+        }
+    }
+
+    self.initCondition = self.initCondition || function () {
+            $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
+            $scope.condition = new ConditionModel($scope.patientUuid)
+            $scope.condition = CommonFunctions.extractUrlArgs(window.location.search)['condition'];
+            $scope.condition = JSON.parse($scope.condition.split("%22").join("\"").split("%20").join(" "));
```

Comment:
What is the above line accomplishing?

Reply:
Can you put it in a properly named function such that i can understand it by looking at the code?

Reply:
ok, i will do that